chain-api/src/utils/signatures.spec.ts

```diff
@@ -358,4 +378,29 @@ describe("signatures", () => {
     // Then
     expect(metamaskPubKey).toEqual(galachainPubKey);
   });
+
+  it("flipped signature should recover to same public key", async () => {
+    // Given
+    const originalSig =
+      "N9aRUvGUedrnOrZch0o0bHUyHHXIUDvV6xOhKsja7j63/eyWDoilWW35iTXFXFQ8uSP3mejoRS4NkVVcd13xchs=";
+    const payload = {
+      firstName: "Tom",
+      id: "1",
+      lastName: "Cruise",
+      photo: "https://jsonformatter.org/img/tom-cruise.jpg"
+    };
+
+    const flippedSignatureObj = flipSignatureParity(normalizeSecp256k1Signature(originalSig));
+    const flippedSig =
+      flippedSignatureObj.r.toString("hex", 32) +
+      flippedSignatureObj.s.toString("hex", 32) +
+      new BN(flippedSignatureObj.recoveryParam === 1 ? 28 : 27).toString("hex", 1);
+
+    // When
+    const originalPubKey = recoverPublicKeyTestFunction(originalSig, payload);
+    const newPubKey = recoverPublicKeyTestFunction(flippedSig, payload);
+
+    // Then
+    expect(originalPubKey).toEqual(newPubKey);
```

Comment:
a little suggestion on comments and the location of `originalPubKey` - the this we are testing here is flipping. It's a bit unreadable in the following diff, but the change is just to reorder some lines and comments.

```suggestion
    const originalPubKey = recoverPublicKeyTestFunction(originalSig, payload);

    // When
    const flippedSignatureObj = flipSignatureParity(normalizeSecp256k1Signature(originalSig));

    // Then
    const flippedSig =
      flippedSignatureObj.r.toString("hex", 32) +
      flippedSignatureObj.s.toString("hex", 32) +
      new BN(flippedSignatureObj.recoveryParam === 1 ? 28 : 27).toString("hex", 1);
    const newPubKey = recoverPublicKeyTestFunction(flippedSig, payload);

    expect(originalPubKey).toEqual(newPubKey);
```